src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/KeyVariantByVariantShard.java

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.VariantShard;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.util.List;
+
+/**
+ * KeyVariantByVariantShard takes a PCollection of variants and keys each read by the variants shards it overlaps.
```

Comment:
keys each variant by the variant shards it overlaps? There are no reads here,right?